extensions/amp-form/0.1/amp-form.js

```diff
@@ -292,11 +295,19 @@ export class AmpForm {
       user().warn(TAG, 'Variable substitutions disabled for non-canonical ' +
           'origin submit action: %s', this.form_);
     }
+
+    for (const k in formObject) {
+      formData['formFields[' + k + ']'] = formObject[k][0];
+    }
+    formData['formId'] = this.form_.id;
+
     if (this.xhrAction_) {
+      this.analyticsEvent_('amp-form-submit', formData);
```

Comment:
variable substitution for input values has not happened yet here. I suggest moving the data preparation and the `analyticsEvent_` call to a private function and call it in `handleXhrSubmit_` and `handleNonXhrGet_` after variable substitution is done.